ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/session-timeout/session-timeout.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import * as React from 'react'
+import styled from '../../styles/styled-components'
+const sessionTimeoutModel = require('../../../component/singletons/session-timeout')
+import { Button, buttonTypeEnum } from '../button'
+import withListenTo, {
+  WithBackboneProps,
+} from '../../container/backbone-container'
+
+const SessionTimeoutRoot = styled.div`
+  height: 100%;
+  width: 100%;
+  display: block;
+  overflow: hidden;
+`
+const Message = styled.div`
+  max-height: calc(100% - 2.25rem);
+  height: auto;
+  text-align: center;
+  padding: ${props => props.theme.mediumSpacing};
+`
+const ButtonStyling = {
+  height: '2.75rem',
+  width: '100%',
+}
+
+type State = {
+  timeLeft: number
+}
+type Props = {} & WithBackboneProps
+
+const renewSession = () => {
+  sessionTimeoutModel.renew()
+}
+
+class SessionTimeout extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props)
+    this.props.listenTo(
+      sessionTimeoutModel,
+      'change:idleTimeoutDate',
+      this.updateState.bind(this)
+    )
+    this.state = {
+      timeLeft: sessionTimeoutModel.getIdleSeconds(),
+    }
+  }
+  componentDidMount() {
+    setInterval(
```

Comment:
I don't think you need `setInterval` now that you're listening to the backbone model now

Reply:
@willwill96 for some reason the `change` event was never getting fired when I was using `listenTo` so I replaced it with `setInterval` entirely. Seeing as the change event should get fired every second as it is I think this will be fine.